src/DynamoCoreWpf/Views/FileTrust/FileTrustWarning.xaml.cs

```diff
@@ -0,0 +1,151 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Windows;
+using System.Windows.Controls;
+using System.Windows.Controls.Primitives;
+using System.Windows.Threading;
+using Dynamo.Controls;
+using Dynamo.Models;
+using Dynamo.ViewModels;
+using Dynamo.Wpf.Controls;
+using Dynamo.Wpf.ViewModels;
+using Dynamo.Wpf.ViewModels.FileTrust;
+
+namespace Dynamo.Wpf.Views.FileTrust
+{
+    /// <summary>
+    /// Interaction logic for RealTimeInfoWindow.xaml
+    /// </summary>
+    public partial class FileTrustWarning : Popup
+    {
+
+        private Window mainWindow;
+        private DynamoViewModel dynViewModel;
+        private FileTrustWarningViewModel fileTrustWarningViewModel;
+
+        public FileTrustWarning(Window dynamoViewWindow)
+        {
+            InitializeComponent();
+
+            mainWindow = dynamoViewWindow;
+            var dynamoView = dynamoViewWindow as DynamoView;
+            if (dynamoView == null) return;
+
+            dynViewModel = dynamoView.DataContext as DynamoViewModel;
+
+            fileTrustWarningViewModel = dynViewModel.FileTrustWViewModel;
+
+            if (fileTrustWarningViewModel == null)
+                fileTrustWarningViewModel = new FileTrustWarningViewModel();
+
+            DataContext = fileTrustWarningViewModel;
+
+            if (dynamoViewWindow == null) return;
+          
+            //Creating the background of the Popup
+            BackgroundRectangle.Rect = new Rect(fileTrustWarningViewModel.PopupBordersOffSet, fileTrustWarningViewModel.PopupBordersOffSet, fileTrustWarningViewModel.PopupRectangleWidth, fileTrustWarningViewModel.PopupRectangleHeight);     
+            
+            fileTrustWarningViewModel.PropertyChanged += ViewModel_PropertyChanged;
+        }
+
+        private void ViewModel_PropertyChanged(object sender, System.ComponentModel.PropertyChangedEventArgs e)
+        {
+            //When the ShowWarningPopup property is changed we need to enable or disable the Dynamo Run section
+            if (e.PropertyName == "ShowWarningPopup" )
```

Comment:
nameof

Reply:
@QilongTang updated to use nameof in the next commit:
https://github.com/DynamoDS/Dynamo/pull/12916/commits/6e3af78f9f27e36771c3ceb54e8ea240bf3c9047